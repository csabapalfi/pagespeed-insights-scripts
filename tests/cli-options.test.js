const {check, parseArgs} = require('../lib/cli-options');

describe('cli-options', () => {
  const url = 'https://www.google.com';
  const mockArgv = (args = []) => ['node', 'pagespeed-score', ...args];

  describe('check', () => {
    it('throws if URL missing', () => {
      expect(() => check({_:mockArgv([])}))
        .toThrow('URL is required');
    });

    it('throws if URL is invalid', () => {
      expect(() => check({_:mockArgv(['oops'])}))
        .toThrow('Invalid URL');
    });

    it('allows if valid URL', () => {
      expect(check({_:mockArgv([url])}))
        .toBeTruthy()
    });

    it('allows if local + lantern-debug', () => {
      expect(check({_:mockArgv([url]), lanternDebug: true, local: true}))
        .toBeTruthy()
    });

    it('allows if local + !lantern-debug', () => {
      expect(check({_:mockArgv([url]), lanternDebug: false, local: true}))
        .toBeTruthy()
    });

    it('allows if !local + !lantern-debug', () => {
      expect(check({_:mockArgv([url]), lanternDebug: false, local: false}))
        .toBeTruthy()
    });

    it('throws if !local + lantern-debug', () => {
      expect(() => check({_:mockArgv([url]), lanternDebug: true, local: false}))
        .toThrow('--lantern-debug only works with --local')
    });
  });

  describe('parseArgs', () => {
    const argv = mockArgv([url]);
    it('returns defaults based on argv', () => {
      const args = parseArgs({argv})
      expect(args).toMatchSnapshot();
    });
  });
});