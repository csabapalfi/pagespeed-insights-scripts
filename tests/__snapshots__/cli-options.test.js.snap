// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli-options parseArgs returns defaults based on argv 1`] = `
Object {
  "lighthouse": Object {
    "cpu-slow-down": 4,
    "cpuSlowDown": 4,
    "enabled": false,
    "module-path": "lighthouse",
    "modulePath": "lighthouse",
  },
  "metrics": Object {
    "benchmark": false,
    "ttfb": false,
    "user-timing-marks": Object {},
    "userTimingMarks": Object {},
  },
  "output": Object {
    "file-prefix": "",
    "filePrefix": "",
    "jsonl": false,
    "lantern-debug": false,
    "lanternDebug": false,
    "save-assets": false,
    "saveAssets": false,
  },
  "runs": 1,
  "strategy": "mobile",
  "url": "https://www.google.com",
  "warmupRuns": undefined,
}
`;
